ckanext/security/plugin/__init__.py

```diff
@@ -27,30 +28,47 @@ class CkanSecurityPlugin(MixinPlugin, p.SingletonPlugin):
     p.implements(p.IActions)
     p.implements(p.IAuthFunctions)
     p.implements(p.ITemplateHelpers)
+    p.implements(p.IValidators, inherit=True)
 
     # BEGIN Hooks for IConfigurer
 
     def update_config(self, config):
         define_security_tables()  # map security models to db schema
 
-        # Monkeypatching all user schemas in order to enforce a stronger
-        # password policy. I tried monkeypatching
-        # `ckan.logic.validators.user_password_validator` instead
-        # without success.
-        core_schema.default_user_schema = \
-            ext_schema.default_user_schema
-        core_schema.user_new_form_schema = \
-            ext_schema.user_new_form_schema
-        core_schema.user_edit_form_schema = \
-            ext_schema.user_edit_form_schema
-        core_schema.default_update_user_schema = \
-            ext_schema.default_update_user_schema
+        # (canada fork only): monkey patching behind config
+        # TODO: upstream contrib??
+        if not tk.asbool(config.get('ckanext.security.use_ivalidators', False)):
+            # Monkeypatching all user schemas in order to enforce a stronger
+            # password policy. I tried monkeypatching
+            # `ckan.logic.validators.user_password_validator` instead
+            # without success.
+            core_schema.default_user_schema = \
+                ext_schema.default_user_schema
+            core_schema.user_new_form_schema = \
+                ext_schema.user_new_form_schema
+            core_schema.user_edit_form_schema = \
+                ext_schema.user_edit_form_schema
+            core_schema.default_update_user_schema = \
+                ext_schema.default_update_user_schema
```

Comment:
Is there a reason to keep the old monkey patching and not use IValidators?

Reply:
@wardi yeah, I don't think so. I don't know why I had the idea to put it behind a config option. It is best to remove it especially if Core adds anything to the user schemas and security does not have it.

For instance, I added that new "cannot modify your own sysadmin privs" to the user schema, and the security extension does not have that in the monkey patch.